python/mxnet/kvstore.py

```diff
@@ -408,6 +408,8 @@ def set_gradient_compression(self, compression_params):
             Other keys in this dictionary are optional and specific to the type
             of gradient compression.
         """
+        if (self.type() == 'device') or ('dist' in self.type()):
```

Comment:
Please pull this file from my updated branch gc-docs. It fixes the mistake here. 

I'm linking the file from that branch here https://github.com/rahul003/mxnet/blob/gc-docs/python/mxnet/kvstore.py